### PR TITLE
feat(server): Support IPv6 addresses for replication

### PR DESCRIPTION
**Pull request by royjacobson**
Second part of fix for #931 

We do DNS resolving for replication commands, and it has to support IPv6.
We try to choose an IPv4 address if possible and then fallback to IPv6 if not.

To check that we manage to replicate a IPv6 address, you can setup a redis docker image
```bash
docker network create --ipv6 --subnet 2001:db8:abcd::/64 test_ipv6
docker run --network test_ipv6 --name my_redis -d redis
docker inspect my_redis | grep GlobalIPv6Address
```
and connect a dragonfly instance to it locally.

### CODE REVIEW

File: src/server/replica.cc [hunk 1]
```diff
@@ -61,21 +61,34 @@ int ResolveDns(std::string_view host, char* dest) {
...
+    CHECK(p->ai_family == AF_INET || p->ai_family == AF_INET6);
+    if (p->ai_family == AF_INET && !found_ipv4) {
       struct sockaddr_in* ipv4 = (struct sockaddr_in*)p->ai_addr;
-      const char* inet_res = inet_ntop(p->ai_family, &ipv4->sin_addr, dest, INET6_ADDRSTRLEN);
-      CHECK_NOTNULL(inet_res);
-      res = 0;
-      break;
+      addr = (void*)&ipv4->sin_addr;
+      CHECK(nullptr != inet_ntop(p->ai_family, addr, ipv4_addr, INET6_ADDRSTRLEN));
+      found_ipv4 = true;
```
Comment (romange) [inline src/server/replica.cc:77]: why not break?

**Comment by romange** [comment]
how did you test this? 
do you know how to setup a Ipv4+IPv6 network (maybe docker compose) where we can check this setup?

**Comment by royjacobson** [comment]
> how did you test this? do you know how to setup a Ipv4+IPv6 network (maybe docker compose) where we can check this setup?

I have an IPv6 loopback (`::1`), so I tested it there. I'm not sure if our testing environment has it

**Comment by royjacobson** [comment]
I added a few lines that start a redis server with an IPv6 address.

File: src/server/replica.cc (hunk 1 above)
Comment (royjacobson) [reply src/server/replica.cc:76]: done

**Review by romange — APPROVED** [review]

**Pull request merged**